scripts/ai/AIDriveStrategyFindBales.lua

```diff
@@ -547,10 +560,25 @@ function AIDriveStrategyFindBales:update(dt)
     AIDriveStrategyFindBales:superClass().update(self, dt)
     self:updateImplementControllers(dt)
 
+    if CpDebug:isChannelActive(self.debugChannel, self.vehicle) then
+        if self.course then
+            self.course:draw()
+        elseif self.ppc:getCourse() then
+            self.ppc:getCourse():draw()
+        end
+    end
+
     if self:areBaleLoadersFull() and self:isReadyToFoldImplements() then
         self:debug('Bale loader is full, stopping job.')
         self.vehicle:stopCurrentAIJob(AIMessageErrorIsFull.new())
     end
+
+    --- Ignores the loaded auto loader bales.
+    --- TODO: Maybe add a delay?
+    local loadedBales = self:getBalesToIgnore()
+    for _, bale in pairs(loadedBales) do 
+        g_baleToCollectManager:lockBaleTemporary(bale)

```

Comment:
Why is this needed? Isn't that bale already locked when we started pathfinding? A few comments to explain when we lock/unlock and why would be useful.